test/success/pi1s1.prl

```diff
@@ -71,3 +71,58 @@ Thm IntSuccEquiv : [
 ] by [
   {`IntSucc, `IntSuccIsEquiv}
 ].
+
+Thm IntSuccPath : [
+  (path [_] (U 0 kan) int int)
+] by [
+  <x> `(V x int int IntSuccEquiv)
+].
+
+Thm S1UnivCover : [
+  (-> S1 (U 0 kan))
+] by [
+  lam x. exact (S1-rec [_] (U 0 kan) x int [x] (@ IntSuccPath x)); refine s1/eq/s1-rec; auto
+].
+
+// Needs rewriting on types
+//
+//Thm S1LoopToInt' : [
+//  (->
+//   [x : S1]
+//   [p : (path [_] S1 base x)]
+//   ($ S1UnivCover x))
+//] by [
+//  lam x, p. `(coe 0~>1 [x] ($ S1UnivCover (@ p x)) 0)
+//].
```

Comment:
So, what's difficult here is to show that `0` is a member of `($ S1UnivCover (@ p 0))`. The most straightforward way is to replace `(@ p 0)` by `base`, but AFAIK we do not have such tactics yet. Internalization would not help unless we have relaxed equality types which do not demand `m in a` for `(= a m m)` to be a type.

It seems we need to improve tactics to easily handle such basic cases. On the other hand, I seemed to figure out a way to work around this for this part of the proof of pi1(S1).

Reply:
Aha! Interesting. So, just to make sure I understand, we have `p` a path in the context whose `0`th endpoint is `base`, and we want to rewrite by this?

I see that you say we would need the Nuprl-style equality type to make it work using internalized equality, but I am having trouble understanding what goes wrong if you just use our equality type as-is. But I can try to understand it by stepping through the proof myself...

Reply:
To clarify, I believe we can also handle this situation by having special tactics that rewrite the types.

Reply:
Hmm. My "workaround" is not good enough; I still need to replace `(@ p 0)` by `base` somewhere.